cmd/collectors/rest/templating.go

```diff
@@ -58,57 +64,230 @@ func (r *Rest) initCache() error {
 	r.prop.instanceLabels = make(map[string]string)
 	r.prop.counters = make(map[string]string)
 
-	for _, c := range counters.GetAllChildContentS() {
+	// private end point do not support * as fields. We need to pass fields in endpoint
+	query := r.Params.GetChildS("query")
+	r.prop.apiType = "public"
+	if query != nil {
+		r.prop.apiType = checkQueryType(query.GetContentS())
+	}
+
+	r.ParseRestCounters(counters, r.prop)
+
+	r.Logger.Info().Strs("extracted Instance Keys", r.prop.instanceKeys).Msg("")
+	r.Logger.Info().Int("count metrics", len(r.prop.metrics)).Int("count labels", len(r.prop.instanceLabels)).Msg("initialized metric cache")
+
+	return nil
+}
+
+func HandleDuration(value string) float64 {
+	// Example: duration: PT8H35M42S
+	timeDurationRegex := `^P(?:(\d+)Y)?(?:(\d+)M)?(?:(\d+)D)?T(?:(\d+)H)?(?:(\d+)M)?(?:(\d+(?:.\d+)?)S)?$`
+
+	regexTimeDuration := regexp.MustCompile(timeDurationRegex)
+	if match := regexTimeDuration.MatchString(value); match {
+		// example: PT8H35M42S   ==>  30942
+		matches := regexTimeDuration.FindStringSubmatch(value)
+		if matches == nil {
+			return 0
+		}
+
+		seconds := 0.0
+
+		//years
+		//months
+
+		//days
+		if matches[3] != "" {
+			f, err := strconv.ParseFloat(matches[3], 64)
+			if err != nil {
+				fmt.Printf("%v", err)
+				return 0
+			}
+			seconds += f * 24 * 60 * 60
+		}
+
+		//hours
+		if matches[4] != "" {
+			f, err := strconv.ParseFloat(matches[4], 64)
+			if err != nil {
+				fmt.Printf("%v", err)
+				return 0
+			}
+			seconds += f * 60 * 60
+		}
+
+		//minutes
+		if matches[5] != "" {
+			f, err := strconv.ParseFloat(matches[5], 64)
+			if err != nil {
+				fmt.Printf("%v", err)
+				return 0
+			}
+			seconds += f * 60
+		}
+
+		//seconds & milliseconds
+		if matches[6] != "" {
+			f, err := strconv.ParseFloat(matches[6], 64)
+			if err != nil {
+				fmt.Printf("%v", err)
+				return 0
+			}
+			seconds += f
+		}
+		return seconds
+	}
+
+	return 0
+}
+
+func HandleTimestamp(value string) float64 {
```

Comment:
looks like a method better suited for templating. Same for all other similiar methods.

Reply:
yes.